### PR TITLE
[CAS-684] Hide MutableLiveDatas in View Models

### PR DESCRIPTION
**Pull request by adasiewiczr**
https://stream-io.atlassian.net/browse/CAS-684

### 🎯 Goal
Hide `MutableLiveData` in view models

### 🛠 Implementation details

1. `MessageInputViewModel::editMessage` is replaced with `MessageInputViewModel::messageToEdit`  and MessageInputViewModel::postMessageToEdit`
2. `MessageInputViewModel::repliedMessage` type is changed to `LiveData` 
3. `MessageListViewModel::mode` type is changed to `LiveData` 
4. I've added a task to update documentation after release

### 🧪 Testing

1. Check if you can edit the message
2. Check if you can reply to the message (after sending the message, it's not rendered properly - I've added a task for that)

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (CMS, cookbooks, tutorial)
- [x] Reviewers added

### 🎉 GIF

giphy

### CODE REVIEW

File: stream-chat-android-docs/src/main/kotlin/io/getstream/chat/docs/cookbook/ui/ChatView.kt [hunk 1]
```diff
@@ -68,7 +68,7 @@ class ChatView : Fragment() {
 
         // Let the message input know when we are editing a message
         messageListView.setMessageEditHandler { message ->
-            messageInputViewModel.editMessage.postValue(message)
+            messageInputViewModel.postMessageToEdit(message)
```
Comment (leandroBorgesFerreira) [inline stream-chat-android-docs/src/main/kotlin/io/getstream/chat/docs/cookbook/ui/ChatView.kt:71]: You can use `messageListView.setMessageEditHandler(messageInputViewModel::postMessageToEdit)`

File: stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/channel/ChannelFragment.kt [hunk 2]
```diff
@@ -79,8 +79,8 @@ class ChannelFragment : Fragment() {
                     is MessageListViewModel.Mode.Normal -> resetThread()
                 }
             }
-            binding.messageListView.setOnMessageEditHandler {
-                editMessage.postValue(it)
+            binding.messageListView.setOnMessageEditHandler { message ->
```
Comment (leandroBorgesFerreira) [inline stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/channel/ChannelFragment.kt:82]: You can use: binding.messageListView.setOnMessageEditHandler(::postMessageToEdit)

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt [hunk 3]
```diff
@@ -158,8 +158,8 @@ class ChatFragment : Fragment() {
                     }
                 }
             }
-            binding.messageListView.setMessageEditHandler {
-                editMessage.postValue(it)
+            binding.messageListView.setMessageEditHandler { message ->
+                postMessageToEdit(message)
```
Comment (leandroBorgesFerreira) [inline stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt:162]: You can use method reference

**Review by leandroBorgesFerreira — APPROVED** [review]
Some small coments

File: UNRELEASED_CHANGELOG.md [hunk 4]
```diff
@@ -51,6 +51,9 @@
 ### ✅ Added
 
 ### ⚠️ Changed
+- Deprecated `MessageInputViewModel::editMessage`. It has `DeprecationLevel.Warning` and still can be used. However, it will be not available in the future, so please consider migrating to use `MessageInputViewModel::messageToEdit` and `MessageInputViewModel::postMessageToEdit` instead.
```
Comment (zsmb13) [inline UNRELEASED_CHANGELOG.md:54]:
```suggestion
- Deprecated `MessageInputViewModel::editMessage`. Use `MessageInputViewModel::messageToEdit` and `MessageInputViewModel::postMessageToEdit` instead.
```

Stronger wording 😉

**Review by zsmb13 — APPROVED** [review]

**Pull request merged**